components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/cli/FetchBuildValidationDataCommand.java

```diff
@@ -148,6 +149,22 @@ private String buildScanIndexToOrdinal(int i) {
         }
     }
 
+    private List<Build> buildsFrom(List<String> runNumsAndBuildScanUrls) {
+        return runNumsAndBuildScanUrls.stream().map(this::buildFrom)
+            .collect(Collectors.toList());
+    }
+
+    private Build buildFrom(String runNumAndBuildScanUrl) {
+        String[] parts = runNumAndBuildScanUrl.split(",");
+        String runNum = parts[0];
+        String buildScanUrl = parts[1];
+        try {
+            return new Build(Integer.parseInt(runNum), new URL(buildScanUrl));
```

Comment:
I would introduce a factory method on the `Build` class that takes two strings.

Reply:
Or possibly even takes the entire comma-separated String.

Reply:
✅ Done